client.go

```diff
@@ -96,6 +96,15 @@ func (c *Client) GetByID(id string) (*session.Session, error) {
 	return s, nil
 }
 
+func (c *Client) DeleteAll() error {
```

Comment:
This function seems to do the same thing as `FlushAll` below, though I can see that it returns just an `error` instead of `*redis.StatusCmd`. I am guessing this is for convenience of getting the error? It makes me wonder if we should hide the details of the `*redis.StatusCmd` type, and not return it from any client function?

Reply:
My thinking was, if we ever switched from go-redis to a different library the DeleteAll() func would still work.

Reply:
Yes that's right, I think that's a good way of doing it. My point was that should we do that for all of the functions on the client? so that changing to a different library would require no function changes?

Reply:
Ah ok, I see what you mean sorry. For some reason I thought I needed all the functions. So they can be removed and the functionality is unchanged?

Reply:
You would still need functions implemented for each, but instead of having the functions that return `*redis.StatusCmd` (like the FlushAll function), you would only have functions like `DeleteAll` that return standard types.

Reply:
Pushed a change, I think these were the functions you were talking about.

Reply:
Yes - I think its better to hide any details of the underlying library. You could still have a ping function for healthchecks, but instead of returning the `*redis.StatusCmd` you could just have `error`. e.g.

```
func (c *Client) Ping() error {
	return c.client.Ping().Err()
}
```

Reply:
Makes sense. I've added it back in and updated the checker in healthcheck.go. Also simplified the code in the `DeleteAll()` func so its basically the same as `Ping()` (I added to much code 😆 )